src/Interpreters/Context.h

```diff
@@ -1276,6 +1278,8 @@ class Context: public ContextData, public std::enable_shared_from_this<Context>
 
     const ServerSettings & getServerSettings() const;
 
+    GgmlModelStorage & getGgmlModelStorage() const;
+
```

Comment:
Why store the model in Context? Is it a caching mechanism to load the model only once? 
Adding more items to the `Context` should be avoided.
Please take a look at the `src/Storages` code path. You can derive from an `ISource` and initialize a model in the c-tor.

Reply:
Edit: please disregard my comment, let's keep it in `Context`.